### PR TITLE
Fix and enhance regression tests

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

* There is a list indexing error when A model was optimized with POT  and B model wasn't optimized with POT and A, B model are in the same task (i.e. Action Classification). So, mark the empty data `"-"` for this cases.
* Enhance the doc-string for each functions.
* Reduce the target data of anomaly tasks (15 --> 3) to reduce the test time

### How to test

After tox -e weekly-test,
```python tests/regression/summarize_test_results.py --output_path .tox/```

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: tests/regression/summarize_test_results.py [hunk 1]
```diff
@@ -55,14 +55,43 @@
...
+    """Get the proper dict item by referencing the index and model information.
+
+    Since all models could be optimized by POT or NNCF, we need to check that there are proper values in the data.
+    For example, if A model could be optimized by both POT and NNCF and B model couldn't be supported by POT and NNCF.
+    In this case, we have POT, NNCF results about A, however, we don't have POT, NNCF results about B.
+
+    So, if we don't have results, we need to mark the empty result as "-".
+
+    """
+    if dict_data:
```
Comment (eunwoosh) [inline tests/regression/summarize_test_results.py]: Current implementation also works but how about to use code as below?
```
if dict_data and len(dict_data) > idx:
    return dict_data[idx][model]
else:
    return "-"
```

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left a minor comment. please take a look.

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left a minor comment. please take a look.

File: tests/regression/summarize_test_results.py (hunk 1 above)
Comment (sungmanc) [reply tests/regression/summarize_test_results.py]: Done. [f1cc084](https://github.com/openvinotoolkit/training_extensions/pull/2095/commits/f1cc084fc482f87c3a4c0b6a11092d0ec66740d8)

**Review by eunwoosh — DISMISSED** [review]
LGTM

**Review by goodsong81 — DISMISSED** [review]

File: tests/regression/summarize_test_results.py [hunk 2]
```diff
@@ -55,14 +55,40 @@
...
 
 
-def get_metric_dict(dict_type, idx: int, model: str):
-    if dict_type:
-        return dict_type[idx][model]
+def get_metric_dict(dict_data: Union[List[Dict[str, Any]], None], idx: int, model: str):
+    """Get the proper dict item by referencing the index and model information.
+
+    Since all models could be optimized by POT or NNCF, we need to check that there are proper values in the data.
+    For example, if A model could be optimized by both POT and NNCF and B model couldn't be supported by POT and NNCF.
```
Comment (sungmanc) [inline tests/regression/summarize_test_results.py]:
```suggestion
    For example, if model A could be optimized by both POT and NNCF and model B couldn't be supported by POT and NNCF.
```

**Review by goodsong81 — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Pull request merged**